1.5/Source/I_Aint_Building_That/HarmonyPatches/AbilityPatch.cs

```diff
@@ -0,0 +1,27 @@
+﻿using HarmonyLib;
+using RimWorld;
+using System.Collections.Generic;
+using Verse;
+
+namespace IAintBuildingThat.HarmonyPatches;
+
+[HarmonyPatch(typeof(Ability), "Initialize")]
+class AbilityPatch
+{
+	static void Postfix(Ability __instance)
+	{
+		var comp = new CompAbilityHide(__instance);
+		if (__instance.comps == null) __instance.comps = [comp];
+		else __instance.comps.Add(comp);
+	}
+}
+
+[HarmonyPatch(typeof(Gizmo), "RightClickFloatMenuOptions", MethodType.Getter)]
+class AbilityRightClickPatch()
+{
+	static void Postfix(Gizmo __instance, ref IEnumerable<FloatMenuOption> __result)
+	{
+		if (__instance is Command_Ability ab) 
+			__result = __result.AddItem(new FloatMenuOption("Hide", () => IAintBuildingThat.settings.HiddenAbilities.Add(ab.Ability)));
```

Comment:
"Hide" should be a translation key, but I've missed a few too as I realised looking at this PR, will do another pass afterward so no worries if you don't remember to translate all the strings, easy to miss some